File: Samples for Solid Edge/[C#]-Samples for Solid Edge/C#/SolidEdge.AddIn.EdgeBar/EdgeBarPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SolidEdge.AddIn.EdgeBar
{
    /// <summary>
    /// Wrapper class for hWnd returned from ISolidEdgeBarEx.AddPageEx().
    /// </summary>
    public class EdgeBarPage : NativeWindow, IDisposable
    {
        private bool _disposed = false;
        private SolidEdgeFramework.SolidEdgeDocument _document;
        private EdgeBarControl _edgeBarControl;

        public EdgeBarPage(IntPtr hWnd, object theDocument, EdgeBarControl edgeBarControl)
            : this(hWnd, (SolidEdgeFramework.SolidEdgeDocument)theDocument, edgeBarControl)
        {
        }

        public EdgeBarPage(IntPtr hWnd, SolidEdgeFramework.SolidEdgeDocument document, EdgeBarControl edgeBarControl)
        {
            _document = document;
            _edgeBarControl = edgeBarControl;

            _edgeBarControl.EdgeBarPage = this;

            // Assign the hWnd to this class.
            AssignHandle(hWnd);

            // Reparent child control to this hWnd.
            NativeMethods.SetParent(_edgeBarControl.Handle, Handle);

            // Show the child control.
            NativeMethods.ShowWindow(_edgeBarControl.Handle, ShowWindowCommands.Maximize);
        }

        ~EdgeBarPage()
        {
            Dispose(false);
        }

        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);      
        }

        public void Dispose(bool disposing)
        {
            if (!_disposed)
            {
                if (disposing)
                {
                    if ((_edgeBarControl != null) && (!_edgeBarControl.IsDisposed))
                    {
                        try
                        {
                            _edgeBarControl.Dispose();
                        }
                        catch
                        {
                        }
                    }
                }

                try
                {
                    ReleaseHandle();
                }
                catch
                {
                }

                _edgeBarControl = null;
                _disposed = true;
            }
        }

        public SolidEdgeFramework.Application Application { get { return _document.Application; } }
        public SolidEdgeFramework.SolidEdgeDocument Document { get { return _document; } }
        public EdgeBarControl EdgeBarControl { get { return _edgeBarControl; } }
    }
}

File: Samples for Solid Edge/[VB]-Samples for Solid Edge/VB/SolidEdge.AddIn.EdgeBar/EdgeBarPage.vb
Imports System.Text

Namespace SolidEdge.AddIn.EdgeBar
	''' <summary>
	''' Wrapper class for hWnd returned from ISolidEdgeBarEx.AddPageEx().
	''' </summary>
	Public Class EdgeBarPage
		Inherits NativeWindow
		Implements IDisposable

		Private _disposed As Boolean = False
		Private _document As SolidEdgeFramework.SolidEdgeDocument
		Private _edgeBarControl As EdgeBarControl

		Public Sub New(ByVal hWnd As IntPtr, ByVal theDocument As Object, ByVal edgeBarControl As EdgeBarControl)
			Me.New(hWnd, CType(theDocument, SolidEdgeFramework.SolidEdgeDocument), edgeBarControl)
		End Sub

		Public Sub New(ByVal hWnd As IntPtr, ByVal document As SolidEdgeFramework.SolidEdgeDocument, ByVal edgeBarControl As EdgeBarControl)
			_document = document
			_edgeBarControl = edgeBarControl

			_edgeBarControl.EdgeBarPage = Me

			' Assign the hWnd to this class.
			AssignHandle(hWnd)

			' Reparent child control to this hWnd.
			NativeMethods.SetParent(_edgeBarControl.Handle, Handle)

			' Show the child control.
			NativeMethods.ShowWindow(_edgeBarControl.Handle, ShowWindowCommands.Maximize)
		End Sub

		Protected Overrides Sub Finalize()
			Dispose(False)
		End Sub

		Public Sub Dispose() Implements IDisposable.Dispose
			Dispose(True)
			GC.SuppressFinalize(Me)
		End Sub

		Public Sub Dispose(ByVal disposing As Boolean)
			If Not _disposed Then
				If disposing Then
					If (_edgeBarControl IsNot Nothing) AndAlso ((Not _edgeBarControl.IsDisposed)) Then
						Try
							_edgeBarControl.Dispose()
						Catch
						End Try
					End If
				End If

				Try
					ReleaseHandle()
				Catch
				End Try

				_edgeBarControl = Nothing
				_disposed = True
			End If
		End Sub

		Public ReadOnly Property Application() As SolidEdgeFramework.Application
			Get
				Return _document.Application
			End Get
		End Property
		Public ReadOnly Property Document() As SolidEdgeFramework.SolidEdgeDocument
			Get
				Return _document
			End Get
		End Property
		Public ReadOnly Property EdgeBarControl() As EdgeBarControl
			Get
				Return _edgeBarControl
			End Get
		End Property
	End Class
End Namespace
